pkg/controller/controller.go

```diff
@@ -485,20 +485,16 @@ func (c *controller) getClusterServiceClassAndClusterServiceBroker(instance *v1b
 
 	}
 
-	authConfig, err := getAuthCredentialsFromClusterServiceBroker(c.kubeClient, broker)
-	if err != nil {
+	brokerClient, found := c.brokerClientManager.BrokerClient(NewClusterServiceBrokerKey(serviceClass.Spec.ClusterServiceBrokerName))
+	if !found {
 		return nil, "", nil, &operationError{
-			reason: errorAuthCredentialsReason,
+			reason: errorNonexistentClusterServiceBrokerReason,
 			message: fmt.Sprintf(
-				"Error getting broker auth credentials for broker %q: %s",
-				broker.Name, err,
+				"The instance references a broker %q which has no OSB client created",
+				serviceClass.Spec.ClusterServiceBrokerName,
```

Comment:
A 'clusterservicebroker' resource is created and at the same time as a 'serviceclass' provisioning request. The provisioning request is being processed before processing adding 'clusterservicebroker'. From the controller perspective (or the implementation of the method) it could happen. Another scenario - deletion of clusterservice broker comes to the controller at the same time as 'deprovisioning' request. Before my PR the problem also could happen. The client can be removed before processing. I'm not aware of every details. If such error occurs, the call will be retried with proper backoff policy.

Reply:
Yes, I realized later that we definitely shouldn't panic.